pysipp/xml/cleaner.py

```diff
@@ -0,0 +1,138 @@
+    doc = imp.createDocument(None, 'scenario', dt)
+
+    new_scen = doc.childNodes[-1]
+    new_scen.writexml = types.MethodType(minidom.monkeypatch_scenario_xml, new_scen)
+
+    in_comment = False
+    for elem in scenario.childNodes:
+        if elem.nodeType == minidom.Node.TEXT_NODE:
+            continue
+        elif elem.nodeType == minidom.Node.CDATA_SECTION_NODE:
+            continue
+        elif elem.nodeType == minidom.Node.ELEMENT_NODE:
+            in_comment = False
+            new_node = process_element(doc, elem)
+            if new_node:
+                new_scen.appendChild(new_node)
+                new_scen.appendChild(doc.createSeperator())
+        elif elem.nodeType == minidom.Node.COMMENT_NODE:
+            comment = elem.data.strip()
+            if in_comment:
+                continue
+            if comment.startswith('In client mode (sipp placing calls),'):
+                in_comment = True
+                continue
+            elif comment.startswith('This delay can be customized'):
+                in_comment = True
+                continue
+            elif comment.startswith("The 'crlf' option inserts a blank"):
+                in_comment = True
+                continue
+            elif comment.startswith('definition of the response time'):
+                in_comment = True
+                continue
+            elif comment.startswith('definition of the call length'):
+                in_comment = True
+                continue
+            elif comment.startswith('By adding rrs="true" (Record'):
+                in_comment = True
+                continue
+            elif comment.startswith('Keep the call open for a while in case the 200'):
+                in_comment = True
+                continue
+            elif comment.startswith("The '[last_*]' keyword is replaced automatically"):
+                in_comment = True
+                continue
+            elif comment.startswith("Packet lost can be simulated in"):
+                in_comment = True
+                continue
+            elif comment.startswith("This program is free software"):
+                in_comment = True
+                continue
+            elif comment.startswith("Handle 200 OK for CANCEL"):
+                in_comment = True
+                continue
+            elif comment.startswith("200 Ok to the CANCEL request OR INVITE request"):
+                in_comment = True
+                continue
+            elif comment.startswith("***************************"):
+                in_comment = True
+                continue
+            in_comment = False
+            new_scen.appendChild(elem)
+        else:
+            in_comment = False
+            new_scen.appendChild(elem)
+
+    # delete the last seperator
+    if new_scen.childNodes and isinstance(new_scen.childNodes[-1], minidom.Newline):
+        del new_scen.childNodes[-1]
+
+    doc.appendChild(new_scen)
+    return doc
+
+
+def main():
```

Comment:
This lacks some documentation about what it does.

It looks like it reformats scenario files, but "Process some sipp scripts" doesn't quite cover that.